packages/astro/src/@types/astro.ts

```diff
@@ -2311,7 +2311,7 @@ export interface AstroIntegration {
 			config: AstroConfig;
 			command: 'dev' | 'build' | 'preview';
 			isRestart: boolean;
-			updateConfig: (newConfig: Record<string, any>) => void;
+			updateConfig: (newConfig: DeepPartial<AstroConfig>) => AstroConfig;
```

Comment:
This certainly is a lot better than before, but it could maybe give a false sense of safety. `{}` is technically `Partial<URL>`, for example.

Reply:
It would be an edge case admittedly, and the runtime logic might actually handle it too. I'll take a closer look when I can, but I'm not blocking a merge.